src/main/scala/devices/debug/Debug.scala

```diff
@@ -289,11 +297,110 @@ class TLDebugModuleOuter(device: Device)(implicit p: Parameters) extends LazyMod
+    val HAWINDOWSELWrDataVal = Wire(init = 0.U(32.W))
+    val HAWINDOWSELWrData = (new HAWINDOWSELFields()).fromBits(HAWINDOWSELWrDataVal)
+    val HAWINDOWSELWrEn   = Wire(init = false.B)
+    val HAWINDOWSELRdEn   = Wire(init = false.B)
+
+    val HAWINDOWRdData = Wire(init = (new HAWINDOWFields()).fromBits(0.U))
+    val HAWINDOWWrDataVal = Wire(init = 0.U(32.W))
+    val HAWINDOWWrData = (new HAWINDOWFields()).fromBits(HAWINDOWWrDataVal)
+    val HAWINDOWWrEn   = Wire(init = false.B)
+    val HAWINDOWRdEn   = Wire(init = false.B)
+
+    if (supportHartArray) {
+      val HAWINDOWSELReset = Wire(init = (new HAWINDOWSELFields().fromBits(0.U)))
+      val HAWINDOWSELNxt = Wire(init = (new HAWINDOWSELFields().fromBits(0.U)))
+      val HAWINDOWSELReg = Wire(init = new HAWINDOWSELFields().fromBits(AsyncResetReg(updateData = HAWINDOWSELNxt.asUInt,
+        resetData = 0,
+        enable = true.B,
+        name = "HAWINDOWSEL"
+      )))
+
+      HAWINDOWSELRdData := Wire(init = HAWINDOWSELReg)
+      HAWINDOWSELNxt := HAWINDOWSELReg
+      when (~dmactive | ~supportHartArray.B) {
+        HAWINDOWSELNxt := HAWINDOWSELReset
+      } .otherwise {
+        when (HAWINDOWSELWrEn) {
+            // Unneeded upper bits of HAWINDOWSEL are tied to 0.  Entire register is 0 if 32 or fewer harts.
+          if (nComponents > 32) {
+            HAWINDOWSELNxt.hawindowsel := HAWINDOWSELWrData.hawindowsel & (log2Up(nComponents) - 5).U
+          } else {
+            HAWINDOWSELNxt.hawindowsel := 0.U
+          }
+        }
+      }
+
+      val numHAMASKSlices = ((nComponents - 1)/32)+1
+      HAWINDOWRdData.maskdata := 0.U     // default, overridden below
+      for (ii <- 0 until numHAMASKSlices) {
+        val SliceMask = if (nComponents > ii*32 + 31) 0xFFFFFFFF  // All harts in this slice exist
+                        else (1<<(nComponents - ii*32)) - 1       // Partial last slice
+        val HAMASKRst = Wire(init = (new HAWINDOWFields().fromBits(0.U)))
+        val HAMASKNxt = Wire(init = (new HAWINDOWFields().fromBits(0.U)))
+        val HAMASKReg = Wire(init = Vec(AsyncResetReg(updateData = HAMASKNxt.asUInt,
+          resetData = 0,
+          enable = true.B,
+          name = "HAMASK")))
+
+        when (ii.U === HAWINDOWSELReg.hawindowsel) {
+          HAWINDOWRdData.maskdata := HAMASKReg.asUInt & SliceMask.U
+        }
+
+        HAMASKNxt.maskdata := HAMASKReg.asUInt
+        when (~dmactive) {
+          HAMASKNxt := HAMASKRst
+        }.otherwise {
+          when (HAWINDOWWrEn && (ii.U === HAWINDOWSELReg.hawindowsel)) {
+            HAMASKNxt.maskdata := HAWINDOWWrData.maskdata
+          }
+        }
+
+         // drive each slice of hamask with HAMASKReg or with new value being written
+        for (jj <- 0 until 32) {
+          if (ii*32 + jj < nComponents) {
+            hamask(ii*32 + jj) := Mux(HAWINDOWWrEn && (ii.U === HAWINDOWSELReg.hawindowsel),
+                                      (HAWINDOWWrData.maskdata >> jj) & 1.U,
```

Comment:
You can do this more clearly by doing `HAWINDOWWrData.maskData.toBools(jj)`

Reply:
```
[error] /scratch/erniee/federation/rocket-chip/src/main/scala/devices/debug/Debug.scala:375:63: overloaded method value toBools with alternatives:
[error]   (implicit sourceInfo: chisel3.internal.sourceinfo.SourceInfo,implicit compileOptions: chisel3.core.CompileOptions)Seq[chisel3.core.Bool] <and>
[error]   ()Seq[chisel3.core.Bool]
[error]  cannot be applied to (Int)
[error]                                       HAWINDOWWrData.maskdata.toBools(jj),
[error]                                                               ^

```

Reply:
Still no luck with this one...the toBools part seems to be working but the problem seems to be the element selector (jj).  The error seems to be saying that a Vec of Bool "cannot be applied" to an Int (meaning jj).  Parens suggested don't change anything.

Reply:
Resolved as suggested in Slack - assign .toBools first to a temp, then index from that.